### PR TITLE
Casdon td pro 2 add local Recip

### PR DESCRIPTION
**Pull request by jianyun8023**
Casdon td pro 2: Add Local recipes

[【PDF】Casdon td pro 2 Local Recipes](https://www.casdon.com.cn/uploads/editor/file/20221124/1669256852863137.pdf)

### CODE REVIEW

File: custom_components/tuya_local/translations/en.json [hunk 1]
```diff
@@ -355,6 +355,42 @@
...
+                "name": "Local Recipe",
+                "state": {
+                    "none": "None",
+                    "steamed_egg_with_okra": "Steamed Egg with Okra",
+                    "steamed_sea_bass": "Steamed Sea Bass",
+                    "steamed_prawns": "Steamed prawns",
+                    "handmade_steamed_bread": "Handmade Steamed Bread",
+                    "fan_steamed_baby_vegetables": "Fan Steamed Baby Vegetables",
+                    "braised_pork": "Braised Pork",
+                    "snow_fungus_and_bird_s_nest": "Snow fungus and bird s nest",
```
Comment (make-all) [inline custom_components/tuya_local/translations/en.json:369]: If ' cannot go into the string, then it is better to remove the space between "bird" and "s". But I think inside "double-quotes", 'apostrophe' should be OK.

File: custom_components/tuya_local/translations/en.json [hunk 2]
```diff
@@ -355,6 +355,42 @@
...
+                    "crab_pot": "Crab Pot",
+                    "potato_ribs": "Potato Ribs",
+                    "coconut_chicken_soup": "Coconut Chicken Soup",
+                    "snack_platter": "Snack Platter",
+                    "chicken_skewers": "Chicken Skewers",
+                    "roasted_pork_knuckle": "Roasted Pork Knuckle",
+                    "dried_lemon": "Dried Lemon",
+                    "pork_shop": "Pork Shop",
```
Comment (make-all) [inline custom_components/tuya_local/translations/en.json:377]: I think maybe this is "Pork chop", not shop (店)

File: custom_components/tuya_local/translations/en.json [hunk 3]
```diff
@@ -355,6 +355,42 @@
                     "celsius": "Celsius",
                     "fahrenheit": "Fahrenheit"
                 }
+            },
+            "oven_local_recipe": {
+                "name": "Local Recipe",
```
Comment (make-all) [inline custom_components/tuya_local/translations/en.json:360]: I think "Built-in" is a closer translation of the Chinese than "Local".
Also, Home Assistant convention is for "recipe" in lower case.

File: custom_components/tuya_local/translations/en.json (hunk 2 above)
Comment (jianyun8023) [reply custom_components/tuya_local/translations/en.json:377]: These English and Chinese words are from the Tuya application. I checked the word, and the original is wrong. It should be `猪肉脯` (pork jerky) instead.

File: custom_components/tuya_local/translations/en.json (hunk 1 above)
Comment (jianyun8023) [reply custom_components/tuya_local/translations/en.json:369]: I added the lost `'`, I'm sorry.

key: `snow_fungus_and_bird_s_nest`
English: `Snow fungus and bird's nest`
Chinese: `雪耳燕窝`

File: custom_components/tuya_local/translations/en.json (hunk 3 above)
Comment (jianyun8023) [reply custom_components/tuya_local/translations/en.json:360]: Indeed, I also believe using “Built-in” is more appropriate. I have done the same in the Chinese context.

**Pull request merged**